index.js

```diff
@@ -197,15 +198,23 @@ function resolveUrlLoader(content, sourceMap) {
           var uri      = initialised.split(/[?#]/g).shift(),
               absolute = uri && findFile.absolute(directory, uri, resolvedRoot);
 
+          var hasQuery = false, querySeperator = '';
```

Comment:
Feel free to merge your new code with the block above if it is more terse.

For example:
```
// split into uri and query/hash and then find the absolute path to the uri
var split    = initialised.split(/([?#])/g), // using a capture group includes the delineator the the output
    uri      = split[0],
    absolute = uri && findFile.absolute(directory, uri, resolvedRoot),
    query    = options.keepQueryInUrl ? split.slice(1).join('') : '';
```
And then later you would have:
```
... .replace(BACKSLASH_REGEX, '/').concat(query) ...
```